### PR TITLE
[templates] remove Directory.Build.targets, add @(ProjectCapability)

### PR DESCRIPTION
**Pull request by jonathanpeppers**
### Description of Change ###

The `Directory.Build.targets` files in the templates were a collection
of workarounds that were needed until .NET MAUI was a workload. When
.NET MAUI was a NuGet package, this was the only way to set required
values for the IDE before NuGet restore. In the early days, you had to
unload/reload MAUI projects after NuGet restore.

We can remove `Directory.Build.targets` from templates now, and
consolidate `@(ProjectCapability)` so this is defined by the workload.

The following capabilities are defined for the IDE:

* `Maui`
* `MauiSingleProject`, `LaunchProfiles`, `XamarinStaticLaunchProfiles`
  when `$(SingleProject)` is `true`

These new capabilities are defined to identify projects (via telemetry)
that bring in portions of .NET MAUI:

* `MauiAssets`
* `MauiBlazor`
* `MauiCore`
* `MauiEssentials`

Other changes:

* `$(_KeepLaunchProfiles)` is set by default when `$(SingleProject)`
  is `true`.
* WinUI workarounds are moved into `WinUI.SingleProject.targets`.
* Fixed `sdk-manifests` casing for `src/DotNet/DotNet.csproj -t:Install`

### PR Checklist ###

- [ ] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?

No

### CODE REVIEW

File: .nuspec/Microsoft.Maui.Controls.SingleProject.targets [hunk 1]
```diff
@@ -60,5 +62,11 @@
   <ItemGroup Condition=" '$(SingleProject)' == 'true' ">
     <ProjectCapability Include="MauiSingleProject" />
     <ProjectCapability Include="LaunchProfiles" />
+    <!--
+      Optional - Enables a list of TFM's and device categories in the debug menu.
+      This allows easily toggling of debug target TFM by selecting the platform.
+      If removed, top level debug targets show as a list of devices for the selected TFM.
+      -->
+    <ProjectCapability Include="XamarinStaticLaunchProfiles" />
```
Comment (jonathanpeppers) [inline .nuspec/Microsoft.Maui.Controls.SingleProject.targets:70]: /cc @joj @tondat this is probably the portion related to the IDE.

@tondat I tried to mark you as a reviewer, but maybe you don't have contributor access on this repo.

**Review by Redth — APPROVED** [review]

File: .nuspec/Microsoft.Maui.Controls.SingleProject.targets (hunk 1 above)
Comment (jonathanpeppers) [reply .nuspec/Microsoft.Maui.Controls.SingleProject.targets:70]: @tondat is suggesting in Teams we might change to:
```xml
<ItemGroup>
  <ProjectCapability Include="MauiSingleProject" />
  <ProjectCapability Include="LaunchProfilesGroupByPlatformFilters"/>
  <ProjectCapability Include="BuildOnlyActiveTargetFrameworkOnDebug" />
</ItemGroup>
```
Also, a target that would improve the developer loop (until we get the appropriate Project System changes);
```xml
<Target Name="ComputeXamarinTargetFrameworks" BeforeTargets="_ComputeTargetFrameworkItems">
  <PropertyGroup>
    <TargetFrameworks Condition="'$(TargetFramework)' == '' AND '$(_SelectedTargetFramework)' != ''">$(_SelectedTargetFramework)</TargetFrameworks>
  </PropertyGroup>
  <Message Text="SelectedTargetFramework = $(_SelectedTargetFramework)" />
  <Message Text="TargetFramework = $(TargetFramework)" />
  <Message Text="TargetFrameworks = $(TargetFrameworks)" />
  <Message Text="BuildingInsideVisualStudio = $(BuildingInsideVisualStudio)" />
</Target>
```
I'll want to test this target a bit, and probably refactor it.

Comment (jonathanpeppers) [reply .nuspec/Microsoft.Maui.Controls.SingleProject.targets:70]: Final capabilities might be:
```xml
<ItemGroup>
  <ProjectCapability Include="MauiSingleProject" />
  <ProjectCapability Include="XamarinStaticLaunchProfiles" Condition=" '$(VisualStudioVersion)' &lt; '17.0' " />
  <ProjectCapability Include="LaunchProfilesGroupByPlatformFilters" Condition=" '$(VisualStudioVersion)' &gt;= '17.0' " />
  <ProjectCapability Include="BuildOnlyActiveTargetFrameworkOnDebug" />
</ItemGroup>
```
`LaunchProfilesGroupByPlatformFilters` is Dev17-only, and you can't have `XamarinStaticLaunchProfiles` at the same time.

**Comment by jonathanpeppers** [comment]
## ComputeXamarinTargetFrameworks target

It does not appear that adding this target will currently work:
```
<Target Name="ComputeXamarinTargetFrameworks" BeforeTargets="_ComputeTargetFrameworkItems">
  <PropertyGroup>
    <TargetFrameworks Condition="'$(TargetFramework)' == '' AND '$(_SelectedTargetFramework)' != ''">$(_SelectedTargetFramework)</TargetFrameworks>
  </PropertyGroup>
  <Message Text="SelectedTargetFramework = $(_SelectedTargetFramework)" />
  <Message Text="TargetFramework = $(TargetFramework)" />
  <Message Text="TargetFrameworks = $(TargetFrameworks)" />
  <Message Text="BuildingInsideVisualStudio = $(BuildingInsideVisualStudio)" />
</Target>
```

This needs to run in the "outer" build when `$(TargetFramework)` is blank and `$(TargetFrameworks)` plural is set. The "outer" build does not even load .NET workloads, those are loaded when `$(TargetFramework)` is set in the "inner" build. I currently do not see a way to provide this target from MAUI.

## ProjectCapability

I think the current logic here is right, but my IDE isn't working as expected. I don't have a way to select my Android emulator, it fails to deploy when I hit the Play button:

image

TBD if this is an IDE issue, or something else related to workloads.

**Comment by jonathanpeppers** [comment]
`$(TargetPlatformIdentifier)` changed to `Android`, proper case, and that caused the IDE issue. See:

* https://github.com/xamarin/xamarin-android/commit/15f23105d69e6c4cc50636b3575e3b14474a7671
* https://github.com/xamarin/XamarinVS/pull/12385

File: .nuspec/Microsoft.Maui.Controls.SingleProject.targets [hunk 2]
```diff
@@ -78,6 +80,10 @@
   <ItemGroup Condition=" '$(SingleProject)' == 'true' ">
     <ProjectCapability Include="MauiSingleProject" />
     <ProjectCapability Include="LaunchProfiles" />
+    <!-- If VS is older than Dev17 -->
+    <ProjectCapability Include="XamarinStaticLaunchProfiles" Condition=" '$(VisualStudioVersion)' != '' and '$(VisualStudioVersion)' &lt; '17.0' " />
+    <!-- Otherwise define LaunchProfilesGroupByPlatformFilters by default -->
+    <ProjectCapability Include="LaunchProfilesGroupByPlatformFilters" Condition=" '$(VisualStudioVersion)' == '' or '$(VisualStudioVersion)' &gt;= '17.0' " />
```
Comment (mattleibow) [inline .nuspec/Microsoft.Maui.Controls.SingleProject.targets:86]: The comment looks backwards? This first check is `&lt;` aka `<`?

Comment (jonathanpeppers) [reply .nuspec/Microsoft.Maui.Controls.SingleProject.targets:86]: I think it's right? `XamarinStaticLaunchProfiles` in Dev16 and `LaunchProfilesGroupByPlatformFilters` in Dev17? @tondat is that correct?

File: Directory.Build.targets [hunk 3]
```diff
@@ -10,20 +10,6 @@
     <PublishTrimmed Condition=" '$(TargetFramework)' == 'net6.0-ios' or '$(TargetFramework)' == 'net6.0-maccatalyst' ">false</PublishTrimmed>
```
Comment (Redth) [inline Directory.Build.targets:10]: @mattleibow can we remove this now?

Comment (mattleibow) [reply Directory.Build.targets:10]: Maybe. I know I did in my pr and I think the issue was that we were using Xcode 12. Now with Xcode 13 all over, we probably can remove it.

Comment (Redth) [inline Directory.Build.targets:10]:
```suggestion
```

File: src/Workload/Microsoft.Maui.Controls.Sdk/Sdk/WinUI.SingleProject.targets [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<!-- Workarounds for WinUI when $(SingleProject) is true -->
+<Project>
+  <!-- Required - Overwrite tasks that are not needed when multitargeting -->
+  <Target Name="ValidateWinUIPlatform" />
+  <Target Name="BinPlaceBootstrapDll" />
```
Comment (mattleibow) [inline src/Workload/Microsoft.Maui.Controls.Sdk/Sdk/WinUI.SingleProject.targets:5]: Does this come in late enough? The WinUI targets from the nugets create this target, and we have to overwrite with a dummy.

Comment (jonathanpeppers) [reply src/Workload/Microsoft.Maui.Controls.Sdk/Sdk/WinUI.SingleProject.targets]: This is imported as late as a workload can do it:
https://github.com/dotnet/maui/blob/f95c2e5c9a6e19817c75ddb93fae98ca8739399f/src/Workload/Microsoft.Maui.Controls.Sdk/Sdk/Microsoft.Maui.Controls.Sdk.After.targets#L9

https://github.com/dotnet/maui/blob/f95c2e5c9a6e19817c75ddb93fae98ca8739399f/src/Workload/Microsoft.Maui.Controls.Sdk/Sdk/Microsoft.Maui.Controls.Sdk.targets#L14-L16

We'd need to test, but if it doesn't work we'd have to put the `Directory.Build.targets` back.

**Review by Redth — APPROVED** [review]

**Review by mattleibow — APPROVED** [review]

**Pull request merged**